levelkdev/swapr-dapp
# Fix expiring quotes

# Summary

Fixes #1255

Add feature to recompute the trades periodically.
Set frequency to 5 minutes.

  # To Test

Go to 'Swap' page
Select token from XDAI to USDC (for example)
Select 'Cow' exchange option
Click on 'Wrap' button
Confirm wrap on Metamask
Click on 'Swap with CoW'
Click on 'Confirm Swap'
Click on 'Sign' (problem: and you will get error message; expected: no error message/successful transaction)

  # Background

COW/Gnosis trades quotes don't come with a TTL. In order to work around this, we will set it on our end.
It doesn't seem to be possible to compute the quote just for the COW trade and cascade that information without computing for all trades. Given that, we periodically compute for all trades.
